main.py

```diff
@@ -88,6 +89,17 @@ def knapsack_bqm(costs, weights, weight_capacity):
 
 sampler = LeapHybridSampler()
 sampleset = sampler.sample(bqm)
-solution = [df['weight'][i] for i in range(len(df['weight'])) if sampleset.record.sample[0][i] == 1.]
-
-print("Found solution {} at energy {}.".format(solution, sampleset.first.energy))
+n_sols = sampleset.record.sample.shape[0]
+for i in range(n_sols):
+
+    # Build solution from returned bitstring
+    solution = []
+    for j0, j in enumerate(sampleset.record.sample[i]):
+        # The x's indicate whether each object has been selected
+        if j == 1. and sampleset.variables[j0].startswith('x'):
+            # Indexing of the weights is different than the bitstring;
+            # cannot guarantee any ordering of the bitstring, but the
+            # weights are numerically sorted
+            a1 = re.search("x(\d+)", sampleset.variables[j0])
```

Comment:
More expressive variable naming please. Namely, what is `a1`?

Also, I am a bit confused at what is being done in this line of code. Isn't `a1` always going to be 0? Since I believe `sampleset.variables[j0] = "x<some number>"`.

Reply:
@m3ller, I was just trying to extract the 772, from 'x772', for example. I got a suggestion to just do a slice, since I know the string always starts with 'x'. I added a line for clarity. Hopefully this helps.